File: src/main/java/com/nbteam/killer/helper/domain/Game.java
package com.nbteam.killer.helper.domain;

import cn.hutool.core.collection.CollUtil;
import cn.hutool.core.util.RandomUtil;
import cn.hutool.core.util.StrUtil;
import lombok.Data;

import java.util.Comparator;
import java.util.HashSet;
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

@Data
public class Game extends GameBook {

    /**
     * 房间号
     */
    String code;

    /**
     * 随机生成4位number作为房间号
     *
     * @return
     */
    public String getCode() {
        if (StrUtil.isBlankOrUndefined(code)) {
            code = String.valueOf(RandomUtil.randomInt(1000, 9999));
        }

        return code;
    }

    /**
     * 随机分发座位号
     */
    public void randomNumber() {
        Set<Integer> numbers = new HashSet<>();
        for (Player p : players) {
            while (true) {
                Integer number = RandomUtil.createSecureRandom(RandomUtil.randomBytes(10)).nextInt(players.size() + 1);// .randomInt(1, rs.length + 1);
                if (!numbers.contains(number) && number > 0) {
                    p.setNumber(number);
                    numbers.add(number);
                    break;
                }
            }
        }

    }

    public List<Player> getPlayers() {
        if (CollUtil.isEmpty(players)) {
            return players;
        }

        return players.stream().sorted(Comparator.comparingInt(Player::getNumber)).collect(Collectors.toList());
    }
}

File: killer-helper-base/src/main/kotlin/com/nbteam/killer/helper/base/domain/Game.kt
package com.nbteam.killer.helper.base.domain

import cn.hutool.core.collection.CollUtil
import cn.hutool.core.util.RandomUtil
import cn.hutool.core.util.StrUtil
import lombok.Data
import java.util.stream.Collectors

@Data
open class Game : GameBook() {
    /**
     * 房间号
     */
    private var code: String? = null

    /**
     * 随机生成4位number作为房间号
     *
     * @return
     */
    fun getCode(): String? {
        if (StrUtil.isBlankOrUndefined(code)) {
            code = RandomUtil.randomInt(1000, 9999).toString()
        }
        return code
    }

    /**
     * 随机分发座位号
     */
    fun randomNumber() {
        val numbers: MutableSet<Int> = HashSet()
        for (p in players) {
            while (true) {
                val number = RandomUtil.createSecureRandom(RandomUtil.randomBytes(10))
                    .nextInt(players.size + 1) // .randomInt(1, rs.length + 1);
                if (!numbers.contains(number) && number > 0) {
                    p.number = number;
                    numbers.add(number)
                    break
                }
            }
        }
    }

    override var players: MutableList<Player>
        get() = if (CollUtil.isEmpty(super.players)) {
            ArrayList();
        } else super.players.stream().sorted(Comparator.comparingInt { obj: Player -> obj.number })
            .collect(Collectors.toList())
        set(players) {
            super.players = players
        }
}